wagn/rails/controllers/card_controller.rb

```diff
@@ -73,6 +73,11 @@ def per_request_setup
     Card::Cache.renew
     Card::Env.reset :controller=>self
     Card::Auth.set_current_from_session
+    
+    if params[:id] && !params[:id].valid_encoding?  # slightly better way to handle encoding issues (than the rescue in load_id)
+                                                    # we should find the place where we produce these bad urls
```

Comment:
I bet they are internal links.  Any ideas about where the third part of that name comes from in the code?  Is it a real cardname when you apply this patched code to it?  The percent encoding decoded should get you a UTF-8 codepoint, right?  UTF encodings are required for URLs, I would think.  I'm guessing it is actually the latin encoding in the patch.